### ISSUE TITLE
API Change vs Select methods and the enableDragging flag

### ISSUE THREAD

**Issue by ripripripriprip** [ANovokmet/svelte-gantt#199]
I may have found a bug where dragging a task updates the Gantt chart differently than selecting a task.

When calling the below after *only* selecting a task:

```
gantt.$set({
    rows: newRows,
    tasks: newTasks,
    from: newFrom,
    to: newTo
})
 ```
 
 The following error occurs: `$rowStore.entities[model.resourceId] is undefined`
 
 It appears that the API drag method will update the `resourceIds` of each task, but only selecting does not...?  If drag is disabled, the bug does not appear.

**Comment by ANovokmet** [comment]
Could you explain in more detail what are you doing after selecting a task? Selecting should not update the `resourceIds` since the resource of the task does not change. If you are using the select event to update the gantt, you could try moving the updates to a `setTimeout`.

See https://github.com/ANovokmet/svelte-gantt/issues/189#issuecomment-1770465105

**Comment by ripripripriprip** [comment]
Here is a reproducible example using your example code and data. Modifications are as follows: 

- I broke the rows and tasks into separate lists
- I added a toggle button which will call a method that calls `gantt.$set...`, switching the rows and tasks of the gantt.

To produce the bug: click a task with the initial gantt, then click the toggle button.  For some reason, if you click _and_ drag a task, _then_ click the toggle button, the bug does not appear.  

Thanks for your time and eyes!

```
import { MomentSvelteGanttDateAdapter, SvelteGantt, SvelteGanttDependencies, SvelteGanttTable } from 'svelte-gantt'
import moment from 'moment'
import { useEffect, useState } from 'react';


const Tree = () => {
    // Loading
    const [loaded, setLoaded] = useState(false)
    const [ganttChart, setGanttChart] = useState(null)

    // Time
    function time(input) {
        return moment(input, 'HH:mm')
    }

    // Start & End
    const currentStart = time('06:00')
    const currentEnd = time('18:00')

    // Time ranges
    const timeRanges = [
        {
            id: 0,
            from: time('10:00'),
            to: time('12:00'),
            classes: null,
            label: 'Lunch',
            resizable: false
        },
        {
            id: 1,
            from: time('15:00'),
            to: time('17:00'),
            classes: null,
            label: 'Dinner'
        }
    ]

    const accountingRows = [
        {
            id: 10,
            label: "Accounting",
            class: 'row-group',
            iconClass: 'fas fa-calculator',
            children: [
                {
                    id: 11,
                    label: "Petunia Mulliner"
                }, {
                    id: 12,
                    label: "Mélina Giacovetti"
                }, {
                    id: 13,
                    label: "Marlène Lasslett"
                }, {
                    id: 14,
                    label: "Adda Youell"
                }
            ]
        }
    ]

    const bussinessDevRows = [
        {
            id: 20,
            label: "Business Development",
            class: 'row-group',
            iconClass: 'fas fa-user-tie',
            children: [
                {
                    id: 21,
                    label: "Pietra Fallow"
                }, {
                    id: 22,
                    label: "Mariellen Torbard"
                }, {
                    id: 23,
                    label: "Renate Humbee"
                }
            ]
        }
    ]

    const accountingTasks = [
         {
                id: 1,
                resourceId: 11,
                label: "LPCVD",
                from: time("9:00"),
                to: time("11:00"),
                classes: "orange"
            }, {
                id: 2,
                resourceId: 12,
                label: "Entrepreneurship",
                from: time("10:00"),
                to: time("12:30"),
                classes: "orange"
            }, {
                id: 3,
                resourceId: 13,
                label: "PET-CT",
                from: time("13:30"),
                to: time("15:00"),
                classes: "orange"
            }, {
                id: 4,
                resourceId: 14,
                label: "Auditing",
                from: time("9:30"),
                to: time("11:30"),
                classes: "orange"
            }
    ]

    const businessDevTasks = [
        {
                id: 5,
                resourceId: 21,
                label: "Security Clearance",
                from: time("15:15"),
                to: time("16:00"),
                classes: "green"
            }, {
                id: 6,
                resourceId: 22,
                label: "Policy Analysis",
                from: time("14:00"),
                to: time("17:00"),
                classes: "blue"
            }, {
                id: 7,
                resourceId: 23,
                label: "Xbox 360",
                from: time("13:30"),
                to: time("14:30"),
                classes: "blue"
            }
    ]


    // Datas to load
    const data = {
        rows: accountingRows,
        tasks: accountingTasks,
        dependencies: []
    }

    // Gantt options
    const options = {
        dateAdapter: new MomentSvelteGanttDateAdapter(moment),
        rows: data.rows,
        tasks: data.tasks,
        dependencies: data.dependencies,
        timeRanges,
        columnOffset: 15,
        magnetOffset: 15,
        rowHeight: 52,
        rowPadding: 6,
        headers: [{ unit: 'day', format: 'MMMM Do' }, { unit: 'hour', format: 'H:mm' }],
        fitWidth: true,
        minWidth: 800,
        from: currentStart,
        to: currentEnd,
        tableHeaders: [{ title: 'Label', property: 'label', width: 140, type: 'tree' }],
        tableWidth: 240,
        ganttTableModules: [SvelteGanttTable],
        ganttBodyModules: [SvelteGanttDependencies]
    }

    // useEffect to load Gantt
    useEffect(() => {
        if (!loaded) {
            // Create Gantt
            setGanttChart(new SvelteGantt({ target: document.getElementById('example-gantt'), props: options }))
            setLoaded(true)
        }
    }, [loaded])

    const toggleData = () => {
        console.log("toggling!")
        ganttChart.$set(
            {
                rows: bussinessDevRows,
                tasks: businessDevTasks
            }
        )
    }

    return (
        <>
            {/* GANTT CONTAINER */}
            <div id="example-gantt"></div>
            <button onClick={toggleData}>toggle</button>
        </>
    )
}


export default Tree;
```

**Comment by ANovokmet** [comment]
Thank you @ripripripriprip. It is fixed in `4.2.4`. 

The issue was that when a task is dragged it is added to a special rendering queue, so when rows and tasks were updated, this 'selected' task was trying to find which resource it belongs to, but in this case it did not exist.
